src/librustc_middle/mir/interpret/mod.rs

```diff
@@ -480,38 +485,38 @@ impl<'tcx> AllocMap<'tcx> {
     /// This function exists to allow const eval to detect the difference between evaluation-
     /// local dangling pointers and allocations in constants/statics.
     #[inline]
-    pub fn get(&self, id: AllocId) -> Option<GlobalAlloc<'tcx>> {
-        self.alloc_map.get(&id).cloned()
+    pub fn get_global_alloc(&self, id: AllocId) -> Option<GlobalAlloc<'tcx>> {
+        self.alloc_map.lock().alloc_map.get(&id).cloned()
     }
 
     /// Panics if the `AllocId` does not refer to an `Allocation`
     pub fn unwrap_memory(&self, id: AllocId) -> &'tcx Allocation {
-        match self.get(id) {
+        match self.get_global_alloc(id) {
             Some(GlobalAlloc::Memory(mem)) => mem,
             _ => bug!("expected allocation ID {} to point to memory", id),
         }
     }
 
     /// Panics if the `AllocId` does not refer to a function
     pub fn unwrap_fn(&self, id: AllocId) -> Instance<'tcx> {
```

Comment:
As a tcx method, this name is not very descriptive. Same for `unwrap_memory`.

What about `unwrap_global_alloc_memory` or so? Something involving `global_alloc`, or at least `global`.

Reply:
I assumed that its argument made that obvious... I was honestly playing with the thought of allowing indexing `TyCtxt` with `AllocId` to get `GlobalAlloc`.

Reply:
The argument type might not be obvious at the call site though.

Reply:
Alternatively, what about `get_global_alloc(...).unwrap_[fn|memory]()`?

Reply:
oh heh, that's nice